sdk/include/opentelemetry/sdk/metrics/state/async_metric_storage.h

```diff
@@ -47,6 +53,11 @@ class AsyncMetricStorage : public MetricStorage, public AsyncWritableMetricStora
     std::lock_guard<opentelemetry::common::SpinLockMutex> guard(hashmap_lock_);
     for (auto &measurement : measurements)
     {
+#ifdef ENABLE_METRICS_EXEMPLAR_PREVIEW
+      exemplar_reservoir_->OfferMeasurement(measurement.second, {}, {},
+                                            std::chrono::system_clock::now());
+#endif
```

Comment:
This makes the async code at par with the sync metric storage, so approved.

In the long term, I think we will need to refactor the exemplar reservoir interface, to reduce overhead.

This calls:

* creates dummy objects, for each measurements
* calls now(), for each measurement
* invoke a virtual function
* only to land on a noop implementation in common cases

Very wasteful compared to representing a noop reservoir with a nullptr, and testing a pointer.

Ok as long as we have a feature flag.